CONTRIBUTING.md

```diff
@@ -199,6 +199,59 @@ The `spimock.New()` function returns an implementation of SPI which mocks the
 underlying SPI driver. The `Flash` object can be tested without hardware
 dependencies.
 
+### VM Tests using QEMU
+For code reading or manipulating hardware, it can be reasonable not to mock out syscalls et al. but to run tests in a virtual environment.
+In case you need to test against certain hardware, you can use a QEMU environment via `pkg/vmtest`. In your package, put the setup and corresponding tests in `vm_test.go`. 
+
+**IMPORTANT Notes**
+* Add `!race` build tag to your `vm_test.go`
+* Setup QEMU inside a usual test function
+* Make sure the tests assuming this setup are skipped in non-VM test run
+* Add your package to `blocklist` in `integration/gotests/gotest_test.go` making sure the test doesn't run over there without the proper QEMU setup.
+
+See below or `pkg/mount/block` for examples.
+```
+//go:build !race
+// +build !race
+
+package foo
+
+import (
+	"github.com/u-root/u-root/pkg/qemu"
+	"github.com/u-root/u-root/pkg/testutil"
+	"github.com/u-root/u-root/pkg/vmtest"
+)
+
+// VM setup:
+//
+//  Describe your setup
+
+func TestVM(t *testing.T) {
+	o := &vmtest.Options{
+		QEMUOpts: qemu.Options{
+			Devices: []qemu.Device{
+				// Configure QEMU here
+				qemu.ArbitraryArgs{...},
+			},
+		},
+	}
+	vmtest.GolangTest(t, []string{"github.com/u-root/u-root/pkg/foo"}, o)
+}
+
+func TestFooRunInQEMU1(t *testing.T) {
+  testutil.SkipIfNotRoot(t)
```

Comment:
Since `testutil` is deprecated, as pointed out further down, what is the alternative here @hugelgupf ? 
`vmtest.SkipWithoutQEMU(t *testing.T)`?

Reply:
pending a better answer, just do the standard
if os.Getuid != 0 {
t.Skip("Skipping since we are not root")
}

or similar?